includes/abstracts/abstract-wc-order.php

```diff
@@ -780,14 +780,38 @@ public function get_item_count( $item_type = '' ) {
 	}
 
 	/**
-	 * Get an order item object, based on it's type.
+	 * Get an order item object, based on it's ID. The item must belong to the current
+	 * order. If the item cannot be found or it doens't belong to current order
```

Comment:
Yeah it's a tough one. IMO this is a fix, or at least, a clarification of the method's intention. Given `get_item()` was only introduced in 3.0, IMO it would be OK to "fix" that in a point release (e.g. 3.0.8). But it's definitely a breaking change.

FWIW, it won't break anything in Subscriptions. We are only using `WC_Abstract_Order::get_item()` when we actually want an item from a specific order. Partly because that's all I thought it did, and partly because we've been using our own `get_item_by_id()` method for the purpose of getting an item by ID without knowing the order it's associated with since way back in Subscriptions v1.2.5.